packages/components/src/Button/Button.tsx

```diff
@@ -31,18 +33,28 @@ export const Button: PolymorphicComponentWithRef<BoxOwnProps, 'button'> =
         },
         ref as RefObject<HTMLSpanElement>
       );
+      const buttonSize = icon ? `button.${size}.icon` : `button.${size}`;
 
       return (
         <Box
           {...buttonProps}
           {...props}
           as={as}
-          variant={[`button.${variant}`, `button.${size}`]}
+          variant={[`button.${variant}`, buttonSize]}
           className={className}
           ref={ref}
         >
           <Box as="span" display="inline-flex" alignItems="center">
-            {children}
+            {icon ? (
+              <>
+                {icon}
+                <Box as="span" pl={size === 'large' ? 'xsmall' : 'xxsmall'}>
```

Comment:
This is not good, since we don't know what values the theme uses for `xmsall` and so on. Could look weird.